question28/js/forgotpassword.js

```diff
@@ -19,21 +19,44 @@ mailInputArea.addEventListener('blur', (e) => {
   toggleDisabledOfSubmitButton(checkAllValidity());
 });
 
-const init = () => {
-  const inputMailData = mailInputArea.value;
+const checkData = (emailAddressValue) => {
   const registeredUserData = JSON.parse(localStorage.getItem('userData'));
+  return ((emailAddressValue === registeredUserData.user_email));
+}
 
-  if (registeredUserData === null) {
-    window.location.href = './notautherize.html';
-    return; 
+const userDataVerification = () => {
+  const userData = mailInputArea.value;
+  return new Promise((resolve,reject) => {
+    if (checkData(userData)){
+      resolve({ token: chance.apple_token(), ok: true, code: 200 });
+    } else {
+      reject({ ok: false, code: 401 }); 
+    }
+  })
+}
+
+const checkExistenceOfUserdata = async() => {
+  try {
+    return await userDataVerification();
+  }
+  catch (e) {
+    console.error(e);
   }
-  if (registeredUserData.user_email === inputMailData) {
-    const tokenForPasswordReset = chance.apple_token();
-    const passwordResetPageUrl = "./register/password.html";
-    window.location.href = `${passwordResetPageUrl}?token=${tokenForPasswordReset}`;
+}
+
+const init = async () => {
+  const verificationResult = await checkExistenceOfUserdata();
+
+  if (!verificationResult) {
+    mailInputArea.nextElementSibling.textContent = "会員登録されていないか入力値が間違っています";
+    submitButton.disabled = true;
     return; 
   }
-  window.location.href = './notautherize.html';
+
+  const tokenForPasswordReset = verificationResult.token;
+  const passwordResetPageUrl = "./register/password.html";
```

Comment:
Q.
should I make` passwordResetPageUrl` variable?
I wrote directly.
I'd really appreciate it if you could enlighten me:)!

Reply:
Thank you for your review!
You are right. I thought it was not necessary to put it in a variable since it would not be used elsewhere. Fixed so please confirm!

fixed commit: https://github.com/csakyo/homework/pull/43/commits/00331682b361c303e83fa7e4196fc3da9ad58c6a